packages/grafana-data/src/field/overrides/processors.ts

```diff
@@ -158,3 +159,27 @@ export interface StatsPickerConfigSettings {
    */
   defaultStat?: string;
 }
+
+interface FieldNamePickerInfoProps {
+  name?: string;
+  field?: Field;
+}
+
+export interface FieldNamePickerConfigSettings {
+  /**
+   * Function is a predicate, to test each element of the array.
+   * Return a value that coerces to true to keep the field, or to false otherwise.
+   */
+  filter?: (field: Field) => boolean;
```

Comment:
This is a non-blocker. I find the function name to be confusing, is it specifically to filter out the type of field, so something like `matchField`?

Reply:
:thinking: maybe -- I was trying to match the style of [Array.filter()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/filter)

Reply:
i always have to double-think about this, too :roll_eyes: